common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/BlockStoreClient.java

```diff
@@ -161,6 +172,18 @@ public void getHostLocalDirs(
       String[] execIds,
       CompletableFuture<Map<String, String[]>> hostLocalDirsCompletable) {
     checkInit();
+    int maxRetries = transportConf.maxIORetries();
+    int retryWaitTime = transportConf.ioRetryWaitTimeMs();
+    boolean enableSaslRetries = transportConf.enableSaslRetries();
+    retry(0, 0, maxRetries, enableSaslRetries, retryWaitTime,
+      () -> getHostLocalDirsInternal(host, port, execIds), hostLocalDirsCompletable);
```

Comment:
> was not to remove the threadpool and replace with single threaded executor

Sorry, I misunderstood you at first.

How about using `Runtime.getRuntime().availableProcessors()` as the core size of ScheduledThreadPool?


Or keep the original logic and keep it consistent with the RetryingBlockTransferor retry logic.

Using CachedThreadPool and Sleep

https://github.com/apache/spark/blob/339d1c9d9d50bce63316ac788626bea998e71b06/common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RetryingBlockTransferor.java#L70-L71

https://github.com/apache/spark/blob/339d1c9d9d50bce63316ac788626bea998e71b06/common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RetryingBlockTransferor.java#L226-L228

Reply:
Since `ScheduledExecutorService` has a fixed core pool size, a large (or small) number of core thread size has their own potentially impacts.
Given how infrequent this is going to be, imo we should use what `RetryingBlockTransferor` does.